keystorage/hashicorp/src/testFixtures/java/tech/pegasys/signers/hashicorp/dsl/HashicorpNode.java

```diff
@@ -12,86 +12,293 @@
-    this.serverTlsCertificate = Optional.ofNullable(serverTlsCertificate);
-  }
+  // initialized after start
+  private HashicorpVaultCommands hashicorpVaultCommands;
+  private Future<ProcessResult> serverFuture;
+  private String vaultToken;
+
+  /**
+   * Create a local Hashicorp Vault process and start it.
+   *
+   * <p>System property vaultInstallDir and vaultBinary should be set before calling this method.
+   *
+   * @param withTls if TLS should be enabled
+   * @return A started instance of HashicorpNode
+   */
+  public static HashicorpNode createAndStartHashicorp(final boolean withTls) {
+    if (vaultInstallDir == null) {
+      throw new IllegalStateException("System property [vaultInstallDir] is missing.");
+    }
+
+    final Path vault = Path.of(vaultInstallDir).resolve(vaultBinary);
+    if (!vault.toFile().exists()) {
+      throw new IllegalStateException("Vault binary doesn't exist. " + vault.toString());
+    }
 
-  public static HashicorpNode createAndStartHashicorp(
-      final DockerClient dockerClient, final boolean withTls) {
     try {
       final HashicorpNode hashicorpNode =
           withTls
-              ? new HashicorpNode(dockerClient, SelfSignedCertificate.generate())
-              : new HashicorpNode(dockerClient);
+              ? new HashicorpNode(vault, SelfSignedCertificate.generate())
+              : new HashicorpNode(vault, null);
       hashicorpNode.start();
       return hashicorpNode;
     } catch (final Exception e) {
       throw new RuntimeException("Failed to create Hashicorp Node.", e);
     }
   }
 
-  private void start() {
-    final HashicorpVaultDockerCertificate hashicorpVaultDockerCertificate =
-        serverTlsCertificate.map(HashicorpVaultDockerCertificate::create).orElse(null);
+  private HashicorpNode(final Path vaultPath, final SelfSignedCertificate serverTlsCertificate) {
+    this.vaultPath = vaultPath;
+    this.serverTlsCertificate = Optional.ofNullable(serverTlsCertificate);
+  }
+
+  void start()
+      throws IOException, CertificateEncodingException, TimeoutException, InterruptedException {
+    startVaultServer();
+    hashicorpVaultCommands = new HashicorpVaultCommands(vaultPath.toString(), vaultUrl());
 
-    hashicorpVaultDocker =
-        HashicorpVaultDocker.createVaultDocker(dockerClient, hashicorpVaultDockerCertificate);
-    Runtime.getRuntime().addShutdownHook(new Thread(this::shutdown));
+    final HashicorpVaultTokens hashicorpVaultTokens = initVault();
+    unseal(hashicorpVaultTokens.getUnsealKey());
+    enableHashicorpKeyValueV2Engine();
   }
 
-  public synchronized void shutdown() {
-    if (hashicorpVaultDocker != null) {
-      hashicorpVaultDocker.shutdown();
-      hashicorpVaultDocker = null;
+  // Start vault server. Sets port variable once started.
+  private void startVaultServer() throws IOException, CertificateEncodingException {
+    LOG.info("Starting vault server");
+    final AtomicBoolean serverStarted = new AtomicBoolean(false);
```

Comment:
probably don't need an atomicBoolean since this is a local var

Reply:
makes it easier for `Awaitility.await().atMost(5, TimeUnit.SECONDS).untilTrue(serverStarted);` which takes an AtomicBoolean. Otherwise I agree, we can use standard boolean as well.